modelling/src/main/java/org/axonframework/modelling/entity/child/EventTargetMatcher.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.modelling.entity.child;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.messaging.unitofwork.ProcessingContext;
+
+/**
+ * Functional interface that determines if a given entity should be targeted for a specific {@link EventMessage}.
+ * Typically used to test whether an entity qualifies to process an event and evolve its state accordingly.
+ *
+ * @param <E> The type of entity this matcher is applied to.
+ * @author Mitchell Herrijgers
+ * @since 5.0.0
+ */
+@FunctionalInterface
+public interface EventTargetMatcher<E> {
```

Comment:
This reminds me of something too: https://github.com/AxonFramework/AxonFramework/blob/master/modelling/src/main/java/org/axonframework/modelling/command/ForwardingMode.java

Reply:
This one is completely fair. I will change the `since` and refer back to the original, in addition to making a note in the docs

Reply:
Cool beans!